model/src/application/component/kind.rs

```diff
@@ -52,6 +78,7 @@ impl ComponentType {
             Self::Button => "Button",
             Self::SelectMenu => "SelectMenu",
             Self::TextInput => "TextInput",
+            Self::Unknown(_) => "Unknown",
```

Comment:
(Commenting on this one, applis on all).
I'm not sure I like the display implementation being "Unknown", should be the number imo

Reply:
This actually isn't possible unless i'm missing something. These functions return an `&'static str`. Converting the number to a string would yield a string that is owned in this scope only and thus can not be returned out of the function.

Reply:
No it's impossible in this method, but I meant for the `Display` implementation; currently it just calls this method, it would be better to print the integer (+ `Unknown`)

Reply:
But we don't have a display implementation for most of these, this looks like the only one that has it. It also would require to duplicate the entire match. It seems better to keep this consistent here, and look at if they should all have display implementations (or if this one should be removed) separately

Reply:
It's as simple as:
```rust
fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
    if let Unknown(int) = self {
        Display::fmt(&int, f)
    } else {
        f.write_str(self.name())
    }
}
```

Reply:
> But we don't have a display implementation for most of these, this looks like the only one that has it.

Ah I missed this, maybe it should be removed at a later point then

Reply:
I like the idea of moving to a `Display` implementation later on.

Reply:
Let's keep it as is for now then to not scope creep this PR, and follow it up to move to display soon after